state-synchronizer/src/coordinator.rs

```diff
@@ -280,20 +289,37 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
                             match event {
                                 Event::NewPeer(peer_id, origin) => {
                                     let peer = PeerNetworkId(network_id, peer_id);
-                                    debug!("[state sync] new peer {}", peer);
+                                    debug!(LogSchema::new(LogEntry::NewPeer)
+                                        .peer(&peer)
+                                        .is_upstream_peer(self.request_manager.is_upstream_peer(&peer, origin))
+                                    );
                                     self.request_manager.enable_peer(peer, origin);
```

Comment:
it is interesting that you have the is_upstream_peer above but regardless you pass it to the request manager

Reply:
ah can probably move new/lost peer logging inside request_manager entirely!